File: tests/typesafety/typescript/exercise_status_misuse.ts
// expect: fails

type Status = "received" | "picked" | "shipped";

function assertNever(value: never): never {
  throw new Error(`unhandled case: ${JSON.stringify(value)}`);
}

function nextAction(status: Status): string {
  switch (status) {
    case "received":
      return "start picking";
    case "picked":
      return "pack the box";
    case "shipped":
      return "send the tracking mail";
    default:
      return assertNever(status);
  }
}

export function handle(): string {
  return nextAction("recieved"); // rejected: not one of the three statuses
}

File: tests/typesafety/python/exercise_status_misuse.py
# expect: fails
# title: The misspelled status, rejected by the checker
# misuse-of: exercise_status_solution
from typing import Literal, assert_never

type Status = Literal["received", "picked", "shipped"]


def next_action(status: Status) -> str:
    match status:
        case "received":
            return "start picking"
        case "picked":
            return "pack the box"
        case "shipped":
            return "send the tracking mail"
        case _:
            assert_never(status)


def handle() -> str:
    return next_action("recieved")
